### PR TITLE
Improve custom slot docs

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- Clarify how custom slot types should be defined and referred to 
- closes https://github.com/RasaHQ/rasa/issues/6692

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by melindaloubser1** [comment]
The automatic review requests happened because I based it on the wrong branch at first (I believe). Please rerequest if I'm wrong.

File: docs/docs/domain.mdx [hunk 1]
```diff
@@ -384,6 +384,36 @@ class NumberOfPeopleSlot(Slot):
         return r
 ```
 
+A custom slot class is implemented as an independent python module,
```
Comment (JEM-Mosig) [inline docs/docs/domain.mdx:387]: The content is good, but can you please use active voice? It'd be great if you could also update the text above to active voice. (See [tone and language guildelines](https://www.notion.so/rasa/Tone-and-language-guidelines-cd96f21292b24c1ab7f7457c7973ec69).)

File: docs/docs/domain.mdx [hunk 2]
```diff
@@ -384,6 +384,36 @@ class NumberOfPeopleSlot(Slot):
...
+Use the module path to refer to the custom slot type in your domain file:
+
+```yaml-rasa title="domain.yaml"
+slots:
+  people:
+    type: addons.my_custom_slots.NumberOfPeopleSlot
+    influence_conversation: true
+```
+
 Rasa Open Source needs some training stories to learn from how to handle the
```
Comment (JEM-Mosig) [inline docs/docs/domain.mdx:417]: I know you didn't add this, but could you rewrite this please? I don't think this is correct english 😅 . Also, we could explain what we mean by "situations".

**Review by JEM-Mosig — COMMENTED** [review]
Looks good! Just needs a change to active voice.

**Review by JEM-Mosig — APPROVED** [review]
🚀

**Pull request merged**